### PR TITLE
Lpd 23651 include the generation of bom release dxp distro in the release

### PR DESCRIPTION
**Pull request by pgydev**

### CODE REVIEW

File: release/_bom.sh [hunk 1]
```diff
@@ -383,4 +461,32 @@ function _manage_bom_jar {
 	fi
 
 	rm -fr jar-temp
+}
+
+function _manage_distro_jar {
+	lc_log DEBUG "Processing release.dxp.distro-'${_PRODUCT_VERSION}-${_BUILD_TIMESTAMP}' for api jar."
+
+	if [[ ! $(echo "${_OSGI_VERSION}" | grep "q") ]]
```
Comment (natocesarrego) [inline release/_bom.sh]: @pgydev, where the variable `_OSGI_VERSION` is being set? Can you please clarify?

File: release/_bom.sh [hunk 2]
```diff
@@ -285,10 +337,36 @@ function generate_poms {
...
+}
+
+function generate_poms_from_scratch {
+	mkdir -p "${_BUILD_DIR}/boms"
+
+	lc_cd "${_BUILD_DIR}/boms"
+
+	rm -f ./*.pom
+	rm -f ./*.jar
+
```
Comment (natocesarrego) [inline release/_bom.sh]: @pgydev, can you replace rebase your code and fix possible conflicts?

Right now on `master`, there is only a method named `generate_poms` since the method `generate_poms_from_scratch` was deleted.

File: release/build_release.sh [hunk 3]
```diff
@@ -121,7 +121,11 @@ function main {
 
 		lc_time_run generate_api_source_jar
 
-		generate_poms
+		lc_time_run download_bnd_files 
+
+		lc_time_run generate_distro_jar
```
Comment (natocesarrego) [inline release/build_release.sh:126]: @pgydev, shouldn't these 2 methods be invoked inside the method `generate_pom_release_dxp_distro` since they are requirements for the creation of the distro bom?

File: release/_bom.sh [hunk 4]
```diff
@@ -89,7 +91,7 @@ function generate_api_jars {
 	do
 		_manage_bom_jar "${module_jar}"
 	done
-} 
+}
```
Comment (natocesarrego) [inline release/_bom.sh:94]: @pgydev, please adjust it in the previous commit

File: release/_bom.sh [hunk 5]
```diff
@@ -10,6 +10,8 @@ function download_bnd_files {
 
 	lc_cd "${_BUILD_DIR}/boms"
 
+	rm -f ./*.jar
```
Comment (natocesarrego) [inline release/_bom.sh:13]: @pgydev, can you move all the changes from the file `_bom.sh` in this commit to the previous commit?

File: release/rebuild_bom_files.sh [hunk 6]
```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
```
Comment (natocesarrego) [inline release/rebuild_bom_files.sh:1]: @pgydev, can you please send the bom rebuild in another pull request?

Thanks

**Review by natocesarrego — CHANGES_REQUESTED** [review]

File: release/_bom.sh [hunk 7]
```diff
@@ -114,6 +127,33 @@ function generate_api_source_jar {
...
 }
 
+function generate_distro_jar {
+	lc_cd "${_BUNDLES_DIR}/tomcat/bin"
+
+	./catalina.sh start
+
+	lc_cd "${_BUILD_DIR}/boms"
+
+	chmod +x biz.aQute.bnd-6.4.0.jar
```
[the remark this answers is not in the document]
Comment (pgydev) [reply release/_bom.sh:139]: @natocesarrego  Of course!

File: release/_bom.sh (hunk 2 above)
Comment (pgydev) [reply release/_bom.sh]: @natocesarrego Yes. I did the same here but i did not rename it and deleted the `generate_poms` because i didn't know for sure that you want to get rid of it.

File: release/build_release.sh (hunk 3 above)
Comment (pgydev) [reply release/build_release.sh:126]: @natocesarrego the `generate_distro_jar` method is not required for the creation of the distro bom. That being said the `download_bnd_files` is required by the `generate_distro_jar` so i should put it in there.

**Review by pgydev — COMMENTED** [review]
I answered the questions and clarified some things
